File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
      int j=0;
      for(int i=0;i<nums.size();i++){
        if(nums[i]!=0){
            swap(nums[i],nums[j]);
            j++;
        }
      }
      int k=0;
      for(int i=0;i<nums.size();i++){
        if(nums[i]!=1){
            swap(nums[i],nums[k]);
            k++;
        }
      }
      int l=0;
      for(int i=0;i<nums.size();i++){
        if(nums[i]!=2){
            swap(nums[i],nums[l]);
            l++;
        }
      }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int j=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]!=0){
                int temp=nums[i];
                nums[i]=nums[j];
                nums[j]=temp;
                j++;
            }
        }
        int k=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]!=1){
                int temp=nums[i];
                nums[i]=nums[k];
                nums[k]=temp;
                k++;
            }
        }
        int l=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]!=2){
                int temp=nums[i];
                nums[i]=nums[l];
                nums[l]=temp;
                l++;
            }
        }
    }
}